PurchasesCoreSwift/Purchasing/OfferingsManager.swift

```diff
@@ -0,0 +1,161 @@
+                                                 offerings: cachedOfferings,
+                                                 error: nil)
+
+        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            if self.deviceCache.isOfferingsCacheStale(isAppBackgrounded: isAppBackgrounded) {
+                Logger.debug(isAppBackgrounded
+                                ? Strings.offering.offerings_stale_updating_in_background
+                                : Strings.offering.offerings_stale_updating_in_foreground)
+
+                self.updateOfferingsCache(appUserID: appUserID,
+                                          isAppBackgrounded: isAppBackgrounded,
+                                          completion: nil)
+
+                Logger.rcSuccess(Strings.offering.offerings_stale_updated_from_network)
+            }
+        }
+    }
+
+    @objc(updateOfferingsCacheWithAppUserID:isAppBackgrounded:completion:)
+    func updateOfferingsCache(appUserID: String, isAppBackgrounded: Bool, completion: ReceiveOfferingsBlock?) {
+        deviceCache.setOfferingsCacheTimestampToNow()
+        operationDispatcher.dispatchOnWorkerThread(withRandomDelay: isAppBackgrounded) {
+            self.backend.getOfferings(appUserID: appUserID) { data, error in
+                if let data = data {
+                    self.handleOfferingsBackendResult(with: data, completion: completion)
+                } else if let error = error {
+                    self.handleOfferingsUpdateError(error, completion: completion)
+                }
+            }
+        }
+    }
+
+}
+
+private extension OfferingsManager {
+
+    func handleOfferingsBackendResult(with data: [String: Any], completion: ReceiveOfferingsBlock?) {
+        let productIdentifiers = extractProductIdentifiers(fromOfferingsData: data)
+
+        productsManager.products(withIdentifiers: productIdentifiers) { products in
+            let productsByID = products.reduce(into: [:]) { result, product in
+                result[product.productIdentifier] = product
+            }
+
+            if let createdOfferings = self.offeringsFactory.createOfferings(withProducts: productsByID, data: data) {
+                self.logProductsMissingIfAppropriate(products: productsByID, offeringsData: data)
+
+                self.deviceCache.cache(offerings: createdOfferings)
+                self.dispatchCompletionOnMainThreadIfPossible(completion,
+                                                              offerings: createdOfferings,
+                                                              error: nil)
+            } else {
+                self.handleOfferingsUpdateError(ErrorUtils.unexpectedBackendResponseError(), completion: completion)
+            }
+        }
+    }
+
+    func handleOfferingsUpdateError(_ error: Error, completion: ReceiveOfferingsBlock?) {
+        Logger.appleError(String(format: Strings.offering.fetching_offerings_error, error as CVarArg))
```

Comment:
we usually use `error.localizedDescription` for these

Reply:
although looking at the obj-c code it looks like we may have missed a call to `localizedDescription` there 🤦